tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -119,69 +118,1165 @@ - (void)dealloc {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            expectedImage = @"150x150.png";
+            [buttonVC textImageStateField].text = @"HS;"; // highlighted-selected
+            [buttonVC textImage].text = expectedImage;
+
+            buttonToTest.highlighted = YES;
+            buttonToTest.selected = YES;
+        });
+        ASSERT_TRUE_MSG(uxEvent->Wait(c_testTimeoutInSec), "FAILED: HighlightedSelected state timed out!");
+
+        expectedImage = @"50x50.png";
+        UIControlCheckStateTransition(buttonToTest,
+                                      uxEvent,
+                                      UIControlStateNormal,
+                                      @selector(currentImage),
+                                      [buttonToTest imageForState:UIControlStateNormal]);
+        expectedImage = @"75x75.png";
+        UIControlCheckStateTransition(buttonToTest,
+                                      uxEvent,
+                                      UIControlStateHighlighted,
+                                      @selector(currentImage),
+                                      [buttonToTest imageForState:UIControlStateHighlighted]);
+        expectedImage = @"125x125.png";
+        UIControlCheckStateTransition(buttonToTest,
+                                      uxEvent,
+                                      UIControlStateSelected,
+                                      @selector(currentImage),
+                                      [buttonToTest imageForState:UIControlStateSelected]);
+    }
+
+    TEST_METHOD(UIButton_BackgroundImageForStateCumulative) {
+        StrongId<UIButtonWithControlsViewController> buttonVC;
+        buttonVC.attach([[UIButtonWithControlsViewController alloc] init]);
+        UXTestAPI::ViewControllerPresenter testHelper(buttonVC);
+
+        UIButton* buttonToTest = [buttonVC defaultButton];
+
+        __block auto uxLayoutEvent = UXEvent::CreateManual();
+        __block StrongId<WXFrameworkElement> layerContent = nil;
+
+        __block auto uxEvent = UXEvent::CreateAuto();
+        __block auto xamlSubscriber = std::make_shared<XamlEventSubscription>();
+        __block NSString* expectedBackgroundImage;
+
+        WXFrameworkElement* xamlElement = [buttonToTest xamlElement];
+        ASSERT_OBJCNE(xamlElement, nil);
+
+        // Wait for the layerContent to be part of the visual tree
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            EventRegistrationToken ert = [xamlElement addLayoutUpdatedEvent:^(RTObject* sender, RTObject* args) {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    // Initially there is no LayerContent element unless we set a background image
+                    if (layerContent == nil) {
+                        // Initially there is no LayerContent element
+                        layerContent = FindXamlChild(xamlElement, @"LayerContent");
```

Comment:
Exactly, the object was released more aggressively under release which made the repro more obvious.